### PR TITLE
Tidy up of Compiler.scala to ensure translate and compile naming conventions are used throughout.

### PR DESCRIPTION
**Pull request by carlpulley-da**
CHANGELOG_BEGIN

  * Tidy up of Compiler.scala to ensure translate and compile naming conventions are used throughout.

CHANGELOG_END

- [x] ensure functions/methods have a return type
- [x] fix spelling errors
- [x] ensure `translate*` methods have a `SExpr0` return type
- [x] ensure `compile*` methods have a `SExpr` return type

### CODE REVIEW

**Review by nickchapman-da — APPROVED** [review]
Nice! thank you

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala [hunk 1]
```diff
@@ -72,13 +72,13 @@ private[lf] object Compiler {
   )
 
   object Config {
-    val Default = Config(
+    val Default: Config = Config(
```
Comment (nickchapman-da) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala:75]: I love that we have types on all the function returns.
But for these `val` types, I don't care so much. You could argue its a bit overkill to say `Config` twice.

Comment (remyhaemmerle-da) [reply daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala:75]: I usually skip the return type for case like that (when the right hand side is a fully applied constructor of a case class).
I do not have strong feeling though.

Comment (carlpulley-da) [reply daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala:75]: After discussion, I think I'll remove these `Config` type annotations - the RHS expression looks suitably stable and, as rightly pointed out, it is not adding any real value here that can be discerned.

**Pull request merged**